src/ensure_plugin/java/me/liliandev/ensure/ensures/EnsureInRange.java

```diff
@@ -0,0 +1,21 @@
+package me.liliandev.ensure.ensures;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.PARAMETER})
+/**
+ * Adds a range check to primitive Numbers
+ * min  is inclusive
+ * max is exclusive
+ */
+public @interface EnsureInRange {
```

Comment:
This isn't currently used right?

Reply:
yes, I wanted to have it just in case. this can ensure nonnegativity with 0 and long max value and a spot in conduit code